src/shared/cloudformation/templateRegistry.ts

```diff
@@ -39,22 +46,23 @@ export class CloudFormationTemplateRegistry extends WatchedFiles<CloudFormation.
         if (template.AWSTemplateFormatVersion || template.Resources) {
             if (template.Transform && template.Transform.toString().startsWith('AWS::Serverless')) {
                 // apply serverless schema
-                await updateYamlSchemasArray(path, 'sam')
+                this.yamlExtension.assignSchema(uri, this.schemas.sam)
             } else {
                 // apply cfn schema
-                await updateYamlSchemasArray(path, 'cfn')
+                this.yamlExtension.assignSchema(uri, this.schemas.standard)
```

Comment:
is this a potential race? we are mutating some state that may be later mutated again?

Reply:
The function call itself is atomic so no. The YAML extension effectively just reads directly from the mapping so there is little room for any race conditions.